woocommerce/woocommerce-ios
# Variations: Remove product attributes

closes #3212

# Why

This PR allows the merchant to remove a product attribute, removing an attribute affects the product's variations in **Core**, so on each attribute update, we need to re-fetch the variations.

Also, when removing the last attribute, we need to show the user the empty state screen to allow them to trigger the first variation flow.

# How

- In `AddAttributeOptionsViewModel`:
  - Added `showMoreButton` property for VC to consume
  - Added `removeCurrentAttribute` that removes the attribute remotely.

- In `AddAttributeOptionsViewController`: 
  - Add "More" button, together with the existing "Next" Button
  - Present action sheet with remove option when tapping the "More" button
  - Present confirmation alert when selecting the remove action
  - Call `ViewModel.removeCurrentAttribute` and notify the parent controller 

- Update `EditAttributesViewController` to provide a `AddAttributeOptionsViewModel` with edit mode.

- Update `ProductVariationsViewModel` to:
  - Add an `shouldShowEmptyState()` method
  - Remove product dependency: Before this PR, there were two sources for the product(VM and VC) that can go out of sync. We need to stick with one(The VC) until the VC is refactored to use `MVVM`

- Update `ProductVariationsViewController` to: 
  - Configure its right button item based on the `VM` 
  - Display or hide the empty state screen when the product attributes are updated
  - Re-fetch variations when the product attributes are updated
  - Make sure the user is on the right screen after deleting the last variation. (EditAttributes after deleting/updating an attribute. VariationsList after deleting the last attribute)
  
  # Demo
  
Delete Any | Delete Last
--- | --- 
![remove-attribute](https://user-images.githubusercontent.com/562080/109062862-7ab8f800-76b6-11eb-87ce-ca631c6a5a8b.gif) | ![remove-last-attribute](https://user-images.githubusercontent.com/562080/109062842-742a8080-76b6-11eb-9062-2450076d078e.gif)

# Testing Steps
- Go to a product with variations and at least two attributes.
- Navigate to the variations list
- Tap on the "more" button and go to edit attributes
- Tap on an attribute
- Tap on the "more" button and see that an action sheet with a "remove" option appears.
- Tap on the "remove" action and see that an alert confirmation appears.
- Tap on the "remove" confirmation and see that you are brought back to the edit attribute screen and that the attribute was removed.
**Repeat process with all other attributes**
- Once the last attribute is removed, see that you are taken back to the variation list screen with the empty state on top of it.

Update release notes:
